src/lib/comms/include/sol-http.h

```diff
@@ -238,7 +239,15 @@ bool sol_http_param_add(struct sol_http_param *params,
     struct sol_http_param_value value) SOL_ATTR_WARN_UNUSED_RESULT;
 void sol_http_param_free(struct sol_http_param *params);
 
-int sol_http_escape_string(char **escaped, const char *value);
+int sol_http_escape_string(char **escaped, const struct sol_str_slice value);
+
+int sol_http_create_uri(char **uri, const char *protocol, const char *server, const char *path, const char *fragment, int port, const struct sol_http_param *params);
```

Comment:
to make it more consistent, should we take all slices?

we can offer a helper:

```c
static inline int
sol_http_create_uri_from_str(char **uri, const char *protocol, const char *server, const char *path, const char *fragment, , int port, const struct sol_http_param *params)
{
    return sol_http_create_uri(uri, sol_str_slice_from_str(protocol), sol_str_slice_from_str(server), sol_str_slice_from_str(path), sol_str_slice_from_str(fragment ? fragment : ""), port, params);
} 
```

this assumes fragment may be `NULL`.